src/views/template-documentation/TemplateDocumentationView.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+
+import InfoIcon from '@material-ui/icons/Info';
+import MenuBookIcon from '@material-ui/icons/MenuBook';
+import ExploreIcon from '@material-ui/icons/Explore';
+import WarningIcon from '@material-ui/icons/Warning';
+import AddCircleOutlineIcon from '@material-ui/icons/AddCircleOutline';
+import PowerSettingsNewIcon from '@material-ui/icons/PowerSettingsNew';
+import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
+import ChevronRightIcon from '@material-ui/icons/ChevronRight';
+
+import { Button } from 'shared/ui';
+
+import { DocumentationSection } from './models';
+
+import csx from './TemplateDocumentationView.scss';
+
+
+const SECTIONS_MOCKED: DocumentationSection[] = [
+  {
+    title: 'Basic informations',
+    icon: <InfoIcon />
+  },
+  {
+    title: 'Introduction',
+    icon: <MenuBookIcon />
+  },
+  {
+    title: 'Setup & Installation',
+    icon: <PowerSettingsNewIcon />
+  },
+  {
+    title: 'Guide',
+    icon: <ExploreIcon />,
+    subSection: ['Architecture', 'Design patterns', 'Components']
+  },
+  {
+    title: 'Issues',
+    icon: <WarningIcon />,
+    subSection: ['Performance in tree list', 'Issue with graph painting', 'Old dependencies']
+  }
+];
+
+const TemplateDocumentationView = () => {
+  const [activeSection, setActiveSection] = useState(0);
+  const [sections, setSections] = useState(SECTIONS_MOCKED);
+
+  const addNewSection = () => {
```

Comment:
You can use useReducer hook for this feature management. Just a tip - will make your code cleaner and also improve rendering performance because dispatcher functions are memoized

Reply:
Few questions about useReducer:

1.  Are there any naming conventions? Like in 'classic' useState - I mean we declare it like this:
const [state, setState] = useState(value) - is there something similar regarding useReducer?

2. Should it be moved outside this component or it should be declared above component, in the same file? If it should be moved, how should I do this? Create a file called 'logic' and move it there, then export in index.ts?

Reply:
1. Basically with userReducer you doing something like that inside component:
 const [entityState, entityDispatcher] = useReducer(initialState); so naming conventions are almost same

2. All logic is moved outside component and created / handler via actions. You can check ref in docs

3. It depends from you only. If file is big you can move this if not can be used from file directly. This is just suggestion - for future